### PR TITLE
Add C# Ice/partial demo that shows how to use partial classes with Ice

### PR DESCRIPTION
**Pull request by pepone**

### CODE REVIEW

File: csharp/Ice/partial/TodoItem.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    {
+        //
+        // All generated constructors call the ice_initialize partial method after
+        // initializing the data members, providing an implementation for ice_initialize
+        // partial method allow us to customize the TodoItem initialization.
+        //
+        partial void ice_initialize()
+        {
+            mId = nextId++;
+            mStatus = Status.Pending;
```
Comment (bernardnormier) [inline csharp/Ice/partial/TodoItem.cs]: I would move this to a data member with a default value in the Slice definition.

File: csharp/Ice/partial/config.client [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#
+# The client reads this property to create the reference to the
+# "ContactDB" object in the server.
```
Comment (bernardnormier) [inline csharp/Ice/partial/config.client]: Should be "todolist" object?

File: csharp/Ice/partial/config.server [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#
+# The server creates one single object adapter with the name
+# "ContactDB". The following line sets the endpoints for this
+# adapter.
```
Comment (bernardnormier) [inline csharp/Ice/partial/config.server]: todolist object again?

**Comment by pepone** [comment]
Updated the demo as suggested

**Pull request merged**